libs/ui/src/hooks/use_devices.ts

```diff
@@ -30,66 +41,49 @@ function getDeviceName(device: KioskBrowser.Device) {
   if (isAccessibleController(device)) {
     return `Accessible Controller (${device.deviceName})`;
   }
-  if (isFujitsuScanner(device)) {
-    return `Fujitsu Scanner (${device.deviceName})`;
+  if (isBatchScanner(device)) {
+    return `Batch Scanner (${device.deviceName})`;
   }
-  if (isPlustekVtm300Scanner(device)) {
-    return `Plustek Scanner (${device.deviceName})`;
+  if (isPrecinctScanner(device)) {
+    return `Precinct Scanner (${device.deviceName})`;
   }
   return `Device (${device.deviceName})`;
 }
 
-export function useHardware({
-  hardware,
-  logger,
-}: UseHardwareProps): UseHardwareResult {
-  const [hasCardReaderAttached, setHasCardReaderAttached] = useState(false);
-  const [
-    hasAccessibleControllerAttached,
-    setHasAccessibleControllerAttached,
-  ] = useState(false);
-  const [hasPlustekScannerAttached, setHasPlustekScannerAttached] = useState(
-    false
-  );
-  const [hasFujitsuScannerAttached, setHasFujitsuScannerAttached] = useState(
-    false
-  );
-  const [hasPrinterAttached, setHasPrinterAttached] = useState(false);
+export function useDevices({ hardware, logger }: Props): Devices {
   const [allDevices, setAllDevices] = useState<KioskBrowser.Device[]>([]);
   const [allPrinters, setAllPrinters] = useState<KioskBrowser.PrinterInfo[]>(
     []
   );
+  const [battery, setBattery] = useState<
+    KioskBrowser.BatteryInfo | undefined
+  >();
   const previousDevices = usePrevious(allDevices);
   const previousPrinters = usePrevious(allPrinters);
 
   useEffect(() => {
     const hardwareStatusSubscription = hardware.devices
       .pipe(map((devices) => Array.from(devices)))
-      .subscribe(async (devices) => {
-        setHasCardReaderAttached(devices.some(isCardReader));
-        setHasAccessibleControllerAttached(
-          devices.some(isAccessibleController)
-        );
-        setHasPlustekScannerAttached(devices.some(isPlustekVtm300Scanner));
-        setHasFujitsuScannerAttached(devices.some(isFujitsuScanner));
-        setAllDevices(devices);
-      });
+      .subscribe(setAllDevices);
 
     const printerStatusSubscription = hardware.printers
       .pipe(map((printers) => Array.from(printers)))
-      .subscribe(async (printers) => {
-        const newHasPrinterAttached = printers.some(
-          ({ connected }) => connected
-        );
-        setHasPrinterAttached(newHasPrinterAttached);
-        setAllPrinters(printers);
-      });
+      .subscribe(setAllPrinters);
+
     return () => {
       hardwareStatusSubscription.unsubscribe();
       printerStatusSubscription.unsubscribe();
     };
   }, [hardware]);
 
+  useInterval(
```

Comment:
I don't think we need to do any logging for the battery but just a heads up for when you add things like printer paper checks to this I think we do need to log when the printer runs out of paper and things like that.

Reply:
noted, thanks!